server/protocol.c

```diff
@@ -1490,8 +1318,59 @@ request_rec *ap_read_request(conn_rec *conn)
 
     ap_run_pre_read_request(r, conn);
 
-    /* Get the request... */
-    if (!read_request_line(r, tmp_bb) || !ap_parse_request_line(r)) {
+    r->request_time = apr_time_now();
+    rv = ap_get_brigade(r->proto_input_filters, tmp_bb, AP_MODE_READBYTES, APR_BLOCK_READ, 0);
```

Comment:
This is a bit quirky,  to read 0 bytes from the input filter to get only the meta buckets. But, if we read any more, we need to push it back somehow so that the request handlers can read it all during processing.